coderd/healthcheck/provisioner.go

```diff
@@ -0,0 +1,141 @@
+package healthcheck
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/mod/semver"
+
+	"github.com/coder/coder/v2/buildinfo"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/coderd/healthcheck/health"
+	"github.com/coder/coder/v2/coderd/provisionerdserver"
+	"github.com/coder/coder/v2/coderd/util/apiversion"
+	"github.com/coder/coder/v2/coderd/util/ptr"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// @typescript-generate ProvisionerDaemonsReport
+type ProvisionerDaemonsReport struct {
+	Severity  health.Severity  `json:"severity"`
+	Warnings  []health.Message `json:"warnings"`
+	Dismissed bool             `json:"dismissed"`
+	Error     *string          `json:"error"`
+
+	ProvisionerDaemons []codersdk.ProvisionerDaemon `json:"provisioner_daemons"`
+}
+
+type ProvisionerDaemonsReportOptions struct {
```

Comment:
These are in an "options" struct but many of them are not optional.

Reply:
This could be called `ProvisionerDaemonsReportDependencies` instead; I was just blindly following convention.